core/primitives/src/network.rs

```diff
@@ -3,6 +3,35 @@ use std::borrow::Borrow;
 use std::fmt;
 use std::hash::{Hash, Hasher};
 use std::net::SocketAddr;
+use std::convert::TryFrom;
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct NodeAddr {
```

Comment:
Should it be `PeerAddr`. Also, what is the distinction between `NodeAddr` and `PeerInfo`?

Reply:
Fixed.